tests/failpoints/cases/test_disk_full.rs

```diff
@@ -0,0 +1,203 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use core::panic;
+use raftstore::store::msg::*;
+use std::time::Duration;
+use test_raftstore::*;
+use tikv_util::config::*;
+
+/*
+Disk full test will have 3 scenarios.
+[leader full]
+1. not allowed: business wirte, region merge/split,etc.
+2. allowed: read, config change(add/remove peer), transfer leader etc.
+
+[minority full]
+1. not allowed: log entry append
+2. allowed: config change,transfer leader,etc.
+
+[majority full]
+1. equal to minority full, no additional design.
+
+*/
+const DISK_FULL_LEADER: &str = "disk_full_leader";
+const DISK_FULL_FOLLOWER_1: &str = "disk_full_follower_1";
+const DISK_FULL_FOLLOWER_2: &str = "disk_full_follower_2";
+
+fn prepare_test_data(cluster: &mut Cluster<ServerCluster>) {
+    let left_key = String::from("5555").into_bytes();
+    let left_value = String::from("55550000").into_bytes();
+    let split_key = String::from("6666").into_bytes();
+    let region = cluster.get_region(b"");
+    cluster.must_split(&region, &split_key);
+    cluster.must_put(&left_key, &left_value);
+    cluster.must_get(&left_key).unwrap();
+    let right_key = String::from("7777").into_bytes();
+    let right_value = String::from("77770000").into_bytes();
+    cluster.must_put(&right_key, &right_value);
+    cluster.must_get(&right_key);
+}
+
+fn fail_split_region(cluster: &mut Cluster<ServerCluster>) {
+    let split_key = String::from("0000").into_bytes();
+    let region = cluster.get_region(&split_key);
+    let split_count_before = cluster.pd_client.get_split_count();
+    cluster.split_region(&region, &split_key, Callback::None);
+    let split_count_after = cluster.pd_client.get_split_count();
+    println!(
+        "fail split count: {} {}",
+        split_count_before, split_count_after
+    );
+    assert!(split_count_before == split_count_after);
+}
+
+fn fail_merge_region(cluster: &mut Cluster<ServerCluster>) {
+    let left_key = String::from("5555").into_bytes();
+    let right_key = String::from("7777").into_bytes();
+    let region1 = cluster.get_region(&left_key);
+    let region2 = cluster.get_region(&right_key);
+    let resp = cluster.try_merge(region1.get_id(), region2.get_id());
+    assert!(is_error_response(&resp));
+}
+
+fn fail_leader_full(cluster: &mut Cluster<ServerCluster>) {
+    fail::cfg(DISK_FULL_LEADER, "return").unwrap();
+
+    let leader = cluster.leader_of_region(1).unwrap();
```

Comment:
It's better to do a manual leader transferring to transfer the leader ship to store 1.